src/storage/src/storage/bidi/builder.rs

```diff
@@ -0,0 +1,361 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::google::storage::v2::BidiReadObjectSpec;
+use crate::model_ext::KeyAes256;
+use crate::read_resume_policy::ReadResumePolicy;
+use crate::request_options::RequestOptions;
+use gaxi::grpc::Client as GrpcClient;
+use gaxi::prost::ToProto;
+
+/// A request builder for `open_object()`.
+///
+/// # Example
+/// ```ignore
+/// use google_cloud_storage::client::Storage;
+/// use google_cloud_storage::retry_policy::RetryableErrors;
+/// async fn sample(client: &Storage) -> anyhow::Result<()> {
+///     # use gax::retry_policy::RetryPolicyExt;
+///     let builder: OpenObject = client
+///         .open_object("projects/_/buckets/my-bucket", "my-object");
+///     let open = builder
+///         .set_generation(123)
+///         .with_retry_policy(
+///             RetryableErrors
+///                 .with_attempt_limit(5)
+///                 .with_time_limit(Duration::from_secs(90)),
+///         )
+///         .send()
+///         .await?;
+///     println!("response details={response:?}");
+///     // Use `open` to read data from `my-object`.
+///     Ok(())
+/// }
+/// ```
+#[derive(Clone, Debug)]
+pub struct OpenObject {
+    spec: BidiReadObjectSpec,
+    options: RequestOptions,
+    client: GrpcClient,
+    reconnect_attempts: u32,
+}
+
+impl OpenObject {
+    pub(crate) fn new(
+        bucket: String,
+        object: String,
+        client: GrpcClient,
+        options: RequestOptions,
+    ) -> Self {
+        let spec = BidiReadObjectSpec {
+            bucket,
+            object,
+            ..BidiReadObjectSpec::default()
+        };
+        Self {
+            spec,
+            options,
+            client,
+            reconnect_attempts: 0_u32,
+        }
+    }
+
+    /// If present, selects a specific revision of this object (as
+    /// opposed to the latest version, the default).
+    pub fn set_generation<T: Into<i64>>(mut self, v: T) -> Self {
```

Comment:
examples for the generation setters?

Reply:
Sigh, maybe this time. I checked them twice.